csv/src/main/scala/akka/stream/alpakka/csv/CsvToMapJavaStage.scala

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.stream.alpakka.csv
+
+import java.nio.charset.Charset
+import java.util.function.Function
+import java.util.stream.Collectors
+import java.{util => ju}
+
+import akka.stream.stage.{GraphStage, GraphStageLogic, InHandler, OutHandler}
+import akka.stream.{Attributes, FlowShape, Inlet, Outlet}
+import akka.util.ByteString
+
+/**
+ * Java API: Converts incoming {@link Collection}<{@link ByteString}> to {@link java.util.Map}<String, ByteString>.
+ *
+ * @param columnNames If given, these names are used as map keys; if not first stream element is used
+ * @param charset Character set used to convert header line ByteString to String
+ */
+class CsvToMapJavaStage(columnNames: Option[ju.Collection[String]], charset: Charset)
```

Comment:
As it is the Java api it should be `Optional` here rather than `Option`

Also, "if you liked it, make it final" ;)